src/main/kotlin/lotto/constants/Rank.kt

```diff
@@ -0,0 +1,27 @@
+package lotto.constants
+
+enum class Rank(val countOfMatch: Int, val winningMoney: Int) {
+    FIRST(6, 2_000_000_000),
+    SECOND(5, 30_000_000),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
+    MISS(0, 0);
+
+    companion object {
+        fun valueOf(countOfMatch: Int, matchBonus: Boolean): Rank {
+            if (matchBonus and (countOfMatch == 5)) {
+                return SECOND
+            }
+            if (countOfMatch in 1..2) {
+                return MISS
+            }
+            return try {
+                values().last { rank -> rank.countOfMatch == countOfMatch }
```

Comment:
```suggestion
                values().lastOrNull { rank -> rank.countOfMatch == countOfMatch }
```
try-catch를 사용하는 것도 좋지만, OrNull 계열의 함수를 사용하는 방법도 있습니다.
엘비스 연산자(?:)를 활용하면 간단하게 null 대신 기본값을 리턴할 수 있습니다.

Reply:
`엘비스 연산자` null 값을 허용하지 않는 변수에 null 값이 들어 갔을 때 null 값을 변환할 수 있는 함수의 결과를 만들어 준다.
`!! 연산자` null 값 보증 ,null 값이 안들어온다는 보증을 해주는 연산자. null 값이 들어오게 되면 오류가 발생한다.

둘이 차이가 없는 연산자로 생각했는데 큰 차이가 있는 연산자였군요..! 수정했습니다. ^__^